apps/aecore/src/aec_tx_pool.erl

```diff
@@ -263,17 +274,28 @@ init([]) ->
     %% The gc db should be owned by this process to ensure that the gc state
     %% is consistent with the actual mempool if any of the servers restart.
     {ok, _GCDb} = pool_db_open(pool_db_gc(), [{keypos, #tx.hash}]),
+    origins_cache_open(origins_cache()),
     GCHeight = top_height(),
     Handled  = ets:new(init_tx_pool, [private]),
     InitF  = fun(TxHash, _) ->
-                     update_pool_on_tx_hash(TxHash, {#dbs{}, GCHeight}, Handled),
+                     update_pool_on_tx_hash(TxHash, {#dbs{}, origins_cache(), GCHeight}, Handled),
                      ok
              end,
     ok = aec_db:ensure_transaction(fun() -> aec_db:fold_mempool(InitF, ok) end),
     ets:delete(Handled),
+    ok = start_origins_cache_gc(),
     lager:debug("init: GCHeight = ~p", [GCHeight]),
     {ok, #state{gc_height = GCHeight}}.
 
+-ifdef(TEST).
+start_origins_cache_gc() ->
+    ok.
+-else.
+start_origins_cache_gc() ->
+    erlang:send_after(rand:uniform(10000), self(), origins_cache_gc),
```

Comment:
So this will run on average every five seconds. Seems often to me.

Perhaps have it run a bit more seldom, and also add an offset for a minimum wait?

EDIT: Ok, I see, this is only the initial call. It would still make sense to add an offset to make it not interfere with the start of the system.

Reply:
Yes, delay sounds reasonable 👍